### PR TITLE
Casmpet 6932 - Can now use helm package command

### PR DESCRIPTION
**Pull request by EthanHolen**
## Summary and Scope

_Summarize what has changed. Explain why this PR is necessary. What is impacted? Is this a new feature, critical bug fix, etc?_

_Is this change backwards incompatible, backwards compatible, or a backwards compatible bugfix?_

## Issues and Related PRs

_List and characterize relationship to Jira/Github issues and other pull requests. Be sure to list dependencies._

* Resolves [issue id](issue link)
* Change will also be needed in `<insert branch name here>`
* Future work required by [issue id](issue link)
* Documentation changes required in [issue id](issue link)
* Merge with/before/after `<insert PR URL here>`

## Testing

_List the environments in which these changes were tested._

### Tested on:

  * `<development system>`
  * Local development environment
  * Virtual Shasta

### Test description:

_How were the changes tested and success verified? If schema changes were part of this change, how were those handled in your upgrade/downgrade testing?_

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)?
- Were continuous integration tests run? If not, why?
- Was upgrade tested? If not, why?
- Was downgrade tested? If not, why?
- Were new tests (or test issues/Jiras) created for this change?

## Risks and Mitigations

_Are there known issues with these changes? Any other special considerations?_

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [ ] Target branch correct
- [ ] CHANGELOG.md updated
- [ ] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: kubernetes/cray-vpa/Chart.yaml
Comment (EthanHolen) [inline kubernetes/cray-vpa/Chart.yaml:1]: Hey @ndavidson-hpe I made updates here to reflect the app and chart version on this page that are wanted https://artifacthub.io/packages/helm/fairwinds-stable/vpa/3.1.0 but I'm not sure if that's right, you also mentioned that this chart needs to pull from upstream. I assume this means that somewhere we have to reference that artifacthub.io page but I'm not sure where that is supposed to be.

File: Jenkinsfile.github [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    options {
+        buildDiscarder(logRotator(numToKeepStr: "10"))
+        timestamps()
+    }
+
+    environment {
+        NAME = "cray-vpa"
+        DESCRIPTION = "Deploys the vpa policy to CSM systems"
+        IS_STABLE = getBuildIsStable()
+        CHART_NAME_1 = "cray-vpa"
```
Comment (ndavidson-hpe) [inline Jenkinsfile.github:17]: Dont need the "_1" theres only one chart
```suggestion
        CHART_NAME = "cray-vpa"
```

File: Jenkinsfile.github [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        CHART_VERSION_1 = getChartVersion(name: env.CHART_NAME_1, isStable: env.IS_STABLE)
```
Comment (ndavidson-hpe) [inline Jenkinsfile.github:18]: Dont need the "_1" theres only one chart
```suggestion
        CHART_VERSION = getChartVersion(name: env.CHART_NAME, isStable: env.IS_STABLE)
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+CHART_NAME_1 ?= cray-vpa
```
Comment (ndavidson-hpe) [inline Makefile:1]: Dont need the "_1" theres only one chart
```suggestion
CHART_NAME ?= cray-vpa
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+CHART_PATH ?= kubernetes
+CHART_VERSION_1 ?= local
```
Comment (ndavidson-hpe) [inline Makefile:3]: Dont need the "_1" theres only one chart
```suggestion
CHART_VERSION ?= local
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+CHART_VERSION_2 ?= local
```
Comment (ndavidson-hpe) [inline Makefile:4]: Dont need a second version
```suggestion

```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+HELM_UNITTEST_IMAGE ?= quintush/helm-unittest:3.3.0-0.2.5
+
+
+all : charts
+charts: chart_setup chart_packages chart_tests
+chart_packages: chart_package_1 chart_package_2
```
Comment (ndavidson-hpe) [inline Makefile:11]: Dont need the "_1" theres only one chart or the second chart
```suggestion
chart_packages: chart_package
```

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+chart_tests: chart_test_1 chart_test_2
```
Comment (ndavidson-hpe) [inline Makefile:12]: Dont need the "_1" theres only one chart or the second chart tests
```suggestion
chart_tests: chart_test
```

File: Makefile [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+chart_setup:
+		mkdir -p ${CHART_PATH}/.packaged
+
+chart_package_1:
```
Comment (ndavidson-hpe) [inline Makefile:17]: Dont need the "_1" theres only one chart
```suggestion
chart_package:
```

File: Makefile [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+		helm dep up ${CHART_PATH}/${CHART_NAME_1}
```
Comment (ndavidson-hpe) [inline Makefile:18]: Dont need the "_1" theres only one chart
```suggestion
		helm dep up ${CHART_PATH}/${CHART_NAME}
```

File: Makefile [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+		helm package ${CHART_PATH}/${CHART_NAME_1} -d ${CHART_PATH}/.packaged --version ${CHART_VERSION_1}
```
Comment (ndavidson-hpe) [inline Makefile:19]: Dont need the "_1" theres only one chart
```suggestion
		helm package ${CHART_PATH}/${CHART_NAME} -d ${CHART_PATH}/.packaged --version ${CHART_VERSION}
```

File: Makefile [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+chart_test_1:
+		helm lint "${CHART_PATH}/${CHART_NAME_1}"
+		docker run --rm -v ${PWD}/${CHART_PATH}:/apps ${HELM_UNITTEST_IMAGE} -3 ${CHART_NAME_1}
```
Comment (ndavidson-hpe) [inline Makefile:23]: Dont need the "_1" theres only one chart
```suggestion
chart_test:
		helm lint "${CHART_PATH}/${CHART_NAME}"
		docker run --rm -v ${PWD}/${CHART_PATH}:/apps ${HELM_UNITTEST_IMAGE} -3 ${CHART_NAME}
```

File: kubernetes/cray-vpa/Chart.yaml [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+apiVersion: v1
+appVersion: "0.14.0"
+description: A Helm chart for Kubernetes
+name: cray-vpa
+version: 3.1.0
```
Comment (ndavidson-hpe) [inline kubernetes/cray-vpa/Chart.yaml:28]: Version should be lower than this in my opinion 
```suggestion
version: 0.0.1
```

File: kubernetes/cray-vpa/Chart.yaml [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+apiVersion: v1
+appVersion: "0.14.0"
```
Comment (ndavidson-hpe) [inline kubernetes/cray-vpa/Chart.yaml:25]: Is this the correct version? Looking at what we suggested what was the chart version of the upstream chart?

**Review by ndavidson-hpe — CHANGES_REQUESTED** [review]

File: kubernetes/cray-vpa/Chart.yaml
Comment (ndavidson-hpe) [reply kubernetes/cray-vpa/Chart.yaml:1]: Once we get the changes done we can work on getting the dependency added. Well include it in this merge but I want to see all the other changes in first.

File: kubernetes/cray-vpa/Chart.yaml (hunk 13 above)
Comment (EthanHolen) [reply kubernetes/cray-vpa/Chart.yaml]: this version would be 3.1.0 instead of the other one where I put it

File: Makefile [hunk 14]
```diff
@@ -0,0 +1,22 @@
+CHART_NAME ?= cray-vpa
+CHART_PATH ?= kubernetes
+CHART_VERSION ?= local
+
+HELM_UNITTEST_IMAGE ?= quintush/helm-unittest:3.3.0-0.2.5
+
+
+all : charts
+charts: chart_setup chart_packages chart_tests
+chart_packages: chart_package chart_package_2
```
Comment (ndavidson-hpe) [inline Makefile:10]: This is failing the build.
```suggestion
charts: chart_setup chart_package chart_tests
```

**Review by ndavidson-hpe — CHANGES_REQUESTED** [review]

**Pull request merged**